src/delfino/click_utils/pass_through_option.py

```diff
@@ -0,0 +1,49 @@
+import shlex
+from typing import List
+
+import click
+
+
+def pass_through_option(command_name: str):
+    """Decorator to add an option for pass-through arguments to a click command.
+
+    This will add a `--{command_name}-option` option to the command. The value
+    for the option should be passed to the underlying external command, which is
+    `command_name`.
+
+    Note, this decorator will not automatically pass the option value to the
+    underlying external command. You need to do it manually.
+
+    Example:
+    ```
+    # Define a command with a pass-through option
+    @click.command()
+    @pass_through_option("mypy")
+    def typecheck(mypy_option: List[str]):
+        run(["mypy", *mypy_option])
+
+    # Call the command with a pass-through option
+    $ delfino typecheck --mypy-option '--ignore-missing-imports --strict'
```

Comment:
Putting the tool name in the option creates an unnecessary cognitive load on the user of delfino. They have to know what the pass option is for every command that uses it (and know that it is available). Could the option be the same for any command? `--pass`, `--pass-options`, `--pass-args` or something similar. Have a look how other command line tools do it.

I encountered this before in form of `--` (double dash), however the support for this in Click is not clear to me:
- https://unix.stackexchange.com/questions/11376/what-does-double-dash-mean
- https://github.com/pallets/click/issues/619
- https://github.com/pallets/click/issues/1340

Maybe @jacksmith15 will have an idea?

```suggestion
    $ delfino typecheck --mypy-option '--ignore-missing-imports --strict'
```

Reply:
Additionally, does the option have to be a string or could it accept variable amount of positional argument? (1..n) Assuming it will be used last, it should not be a problem that it captures everything after this option.

Example:

```shell script
delfino typecheck --mypy-option --ignore-missing-imports --strict
```

Reply:
> I see an argument that a command could run more tools in the background. And therefore there could be multiple options to pass to different tools. But I think support for that is a premature optimisation. We barely have a use case for passing args to a single tool.

👍 let's not think about supporting multiple tools.

> Putting the tool name in the option creates an unnecessary cognitive load on the user of delfino. They have to know what the pass option is for every command that uses it (and know that it is available). Could the option be the same for any command? --pass, --pass-options, --pass-args or something similar. Have a look how other command line tools do it.

Using the same name is possible but use's still have to know which tool the option values to be passed. For example a user has to know `--pass` option to be passed to `pytest` for `test-unit`. Given the name `test-unit` doesn't expose the underlying tooling name, `pytest`, isn't it easier to understand have tooling name in a option name as `--pytest-pass`?

> Have a look at how other command line tools do it.

I think `--` is the most common way to do this. I've taken look at the issues you provide but I think it works in our use case.

`delfino typecheck --mypy-option --ignore-missing-imports --strict` Is not possible as [multi-value options](https://click.palletsprojects.com/en/8.1.x/options/#multi-value-options) does only support fixed number of args.

Reply:
We've discussed offline as https://github.com/radeklat/delfino/issues/34#issuecomment-1256142952.

And I've implemented [55f9630](https://github.com/radeklat/delfino/pull/35/commits/55f96305e2e75d46039e05589fe940e20f34df4c).